File: 151004/Bordilovsky/DC_NoSQL/Publisher/Services/Mappers/NoteMapper.cs
using AutoMapper;
using DC_REST.DTOs.Request;
using DC_REST.DTOs.Response;
using DC_REST.Entities;

namespace DC_REST.Services.Mappers
{
	public class NoteMapper: Profile
	{
		public NoteMapper() 
		{
			CreateMap<Note, NoteResponseTo>();
			CreateMap<NoteRequestTo, Note>();
		}
	}
}

File: 151003/Savitski/publisher/src/main/java/com/example/distributedcomputing/mapper/NoteMapper.java
package com.example.distributedcomputing.mapper;

import com.example.distributedcomputing.model.entity.Note;
import com.example.distributedcomputing.model.request.NoteRequestTo;
import com.example.distributedcomputing.model.response.NoteResponseTo;
import org.mapstruct.Mapper;

import java.util.List;

@Mapper(componentModel = "spring")
public interface NoteMapper {
    Note dtoToEntity(NoteRequestTo noteRequestTo);
    List<Note> dtoToEntity(Iterable<NoteRequestTo> notes);

    NoteResponseTo entityToDto(Note note);

    List<NoteResponseTo> entityToDto(Iterable<Note> notes);
}
